src/test/java/de/tum/in/www1/artemis/metis/PostIntegrationTest.java

```diff
@@ -525,6 +564,43 @@ public void testGetPostsForCourse_WithLectureIdRequestParam() throws Exception {
         assertThat(returnedPosts).hasSameSizeAs(existingLecturePosts);
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testGetPostsForCourse_WithPlagiarismCaseIdRequestParam_asInstructor() throws Exception {
+        // request param plagiarismCaseId will fetch all posts that match this context filter
+        var params = new LinkedMultiValueMap<String, String>();
+        params.add("plagiarismCaseId", plagiarismCaseId.toString());
+
+        List<Post> returnedPosts = request.getList("/api/courses/" + courseId + "/posts", HttpStatus.OK, Post.class, params);
+        database.assertSensitiveInformationHidden(returnedPosts);
+        // get amount of posts with certain plagiarism context
+        assertThat(returnedPosts).hasSameSizeAs(existingPlagiarismPosts);
+    }
+
+    @Test
+    @WithMockUser(username = "student1", roles = "USER")
+    public void testGetPostsForCourse_WithPlagiarismCaseIdRequestParam_asStudent() throws Exception {
+        // request param plagiarismCaseId will fetch all posts that match this context filter
+        var params = new LinkedMultiValueMap<String, String>();
+        params.add("plagiarismCaseId", plagiarismCaseId.toString());
+
+        List<Post> returnedPosts = request.getList("/api/courses/" + courseId + "/posts", HttpStatus.OK, Post.class, params);
+        database.assertSensitiveInformationHidden(returnedPosts);
+        // get amount of posts with certain plagiarism context
+        assertThat(returnedPosts).hasSameSizeAs(existingPlagiarismPosts);
+    }
+
+    @Test
+    @WithMockUser(username = "student2", roles = "USER")
+    public void testGetPostsForCourse_WithPlagiarismCaseIdRequestParam_asStudent_Forbidden() throws Exception {
```

Comment:
Should be forbidden because the notification is not related to `student2`, right?